lib/valkyrie/derivative_service.rb

```diff
@@ -0,0 +1,27 @@
+# frozen_string_literal: true
+require 'hydra/derivatives'
+module Valkyrie
+  class DerivativeService
+    Hydra::Derivatives.source_file_service = Valkyrie::LocalFileService
+    Hydra::Derivatives.output_file_service = Valkyrie::PersistDerivatives
```

Comment:
Hydra-derivatives will read from a tmp file on disk and write to a tmp file on disk, which is then streamed into the storage adapter and removed. I did it this way largely because I didn't understand how to do it more directly.